src/context/PatronDataContext.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState, createContext, type ReactNode } from "react"
+import type {
+  PatronDataContextType,
+  MyAccountPatronData,
+} from "../types/myAccountTypes"
+import { BASE_URL } from "../config/constants"
+/**
+ * Wrapper context component that controls state for the PatronData component
+ */
+export const PatronDataContext = createContext<PatronDataContextType | null>(
+  null
+)
+
+export const PatronDataProvider = ({
```

Comment:
This is great and make sense this is needed for this section of RC.

This is not a necessary change but just want to show a pattern I like to use which is to create a custom hook in this same file:

```
function usePatronDataContext() {
  const context = useContext(PatronDataContext);
  if (!context) {
    console.warn("usePatronDataContext must be used within PatronDataProvider");
    // throw error ?
  } 
  return context;
}
```

This context will only be used in `/account` so there's no worry about not having the provider, but it's a general pattern you might find in other react apps. This also reduces importing both `useContext` and `PatronDataContext` into every component where you need patron data.

```
const { updatedAccountData } = usePatronDataContext();
```

I'm not sure if this will make the tests any easier or more complicated. Try it and if it's not worth it, then scrap it.

Reply:
I'm going to skip this for now.